### PR TITLE
feat: add support to asset type "thing"

### LINKED ISSUE #185: [Feature] Subpage: Edge Computing
## Motivation / Problem

We need to communicate the Compute-to-Edge paradigm on the MVG and make it visible.

## Solution

We should introduce a new subpage **Edge Computing**, which
- explains the CtE concept in a brief section (content will be added to the issue)
- lists relevant things as new `edge` asset teaser type

## Additional Information:
- Query aquarius for assets with `metadata.attributes.main.type = thing` and display with AssetTeaser:
  - display new type ‘Edge Device’ (instead of 'dataset' | ' algorithm'), icon see below
  - title as currently
  - network as currently
  - no price
  - no description, instead:
    - show device model `findServiceByType('edge').provider.device.model`
    - show number of available assets `findServiceByType('edge').attributes.main.availableServices.length`

In the first iteration we do not need asset details page for edge devices → disable teasers to be not clickable

--- 
Icon for the new type 
```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path fill="currentColor" d="M30 19h-4v-4h-2v9H8V8h9V6h-4V2h-2v4H8a2.002 2.002 0 0 0-2 2v3H2v2h4v6H2v2h4v3a2.002 2.002 0 0 0 2 2h3v4h2v-4h6v4h2v-4h3a2.003 2.003 0 0 0 2-2v-3h4Z"/><path fill="currentColor" d="M21 21H11V11h10zm-8-2h6v-6h-6zm18-6h-2A10.012 10.012 0 0 0 19 3V1a12.013 12.013 0 0 1 12 12z"/><path fill="currentColor" d="M26 13h-2a5.006 5.006 0 0 0-5-5V6a7.008 7.008 0 0 1 7 7Z"/></svg>
```

### PR DESCRIPTION
**Pull request by LucaMilanese90**
Fixes #185 #198 #200

## Proposed Changes

  - add `edge devices` filter in search page
  - show details and available devices in the asset details page
  - disable ctd button if the edge device is unavailable

### CODE REVIEW

File: src/components/templates/Page.tsx [hunk 1]
```diff
@@ -30,6 +30,7 @@ export interface PageProps {
   description?: string
   noPageHeader?: boolean
   headerCenter?: boolean
+  isEdgeNetwork?: boolean
```
Comment (moritzkirstein) [inline src/components/templates/Page.tsx:33]: suggestion(blocking): I think `isEdgeNetwork` is a bit confusing. Maybe renaming this to `isEdgeProvider` makes more sense here? Would apply to all occurences.

Comment (moritzkirstein) [inline src/components/templates/Page.tsx:33]: suggestion(blocking): I think `isEdgeNetwork` is a bit confusing. Maybe renaming this to `isEdgeProvider` makes more sense here? (Applies to all occurences)

**Review by moritzkirstein — CHANGES_REQUESTED** [review]

File: src/components/organisms/AssetContent/EdgeDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        <MetaItem
+          title="Assets on this device"
+          content={
+            <Table
+              columns={columns}
+              data={availableAssets}
+              isLoading={isLoading}
+              emptyMessage="No available assets on this device."
+              noTableHead
+            />
```
[the remark this answers is not in the document]
Comment (LucaMilanese90) [reply src/components/organisms/AssetContent/EdgeDetails.tsx:91]: Nice idea, I like it!

File: src/components/organisms/AssetContent/index.tsx [hunk 3]
```diff
@@ -71,6 +74,11 @@ export default function AssetContent(props: AssetContentProps): ReactElement {
   const [serviceSelfDescription, setServiceSelfDescription] = useState<string>()
   const { appConfig } = useSiteMetadata()
 
+  const showEdgeDetails = useMemo(
+    () => !!(ddo as EdgeDDO)?.findServiceByType('edge'),
```
[the remark this answers is not in the document]
Comment (LucaMilanese90) [reply src/components/organisms/AssetContent/index.tsx:78]: You mean for the double `!!`?

[the remark this answers is not in the document]
Comment (LucaMilanese90) [reply src/components/organisms/AssetContent/index.tsx:78]: What if I just explicitly declare the type of the variable? Would that help in not needing the comment?

`const showEdgeDetails: boolean = useMemo(
    () => !!(ddo as EdgeDDO)?.findServiceByType('edge'),
    [ddo]
  )`

[the remark this answers is not in the document]
Comment (moritzkirstein) [reply src/components/organisms/AssetContent/index.tsx:77]: I think it helps - let's use this for now

**Review by moritzkirstein — APPROVED** [review]

**Pull request merged**